test/e2e/healthcheck_universal_test.go

```diff
@@ -0,0 +1,64 @@
+package e2e_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/kumahq/kuma/pkg/config/core"
+	. "github.com/kumahq/kuma/test/framework"
+)
+
+var _ = Describe("Test application HealthCheck on Universal", func() {
+
+	var cluster Cluster
+	var deployOptsFuncs []DeployOptionsFunc
+
+	BeforeEach(func() {
+		cluster = NewUniversalCluster(NewTestingT(), Kuma1, Verbose)
+		deployOptsFuncs = []DeployOptionsFunc{}
+
+		err := NewClusterSetup().
+			Install(Kuma(core.Standalone, deployOptsFuncs...)).
+			Setup(cluster)
+		Expect(err).ToNot(HaveOccurred())
+		err = cluster.VerifyKuma()
+		Expect(err).ToNot(HaveOccurred())
+
+		echoServerToken, err := cluster.GetKuma().GenerateDpToken("default", "echo-server_kuma-test_svc_8080")
+		Expect(err).ToNot(HaveOccurred())
+		demoClientToken, err := cluster.GetKuma().GenerateDpToken("default", "demo-client")
+		Expect(err).ToNot(HaveOccurred())
+
+		err = EchoServerUniversal("universal", "default", echoServerToken, ProxyOnly())(cluster)
+		Expect(err).ToNot(HaveOccurred())
+		err = DemoClientUniversal("default", demoClientToken)(cluster)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		if ShouldSkipCleanup() {
+			return
+		}
+		Expect(cluster.DeleteKuma(deployOptsFuncs...)).To(Succeed())
+		Expect(cluster.DismissCluster()).To(Succeed())
+	})
+
+	It("should update dataplane.inbound.health", func() {
+		Eventually(func() (string, error) {
+			output, err := cluster.GetKumactlOptions().RunKumactlAndGetOutputV(Verbose, "get", "dataplane", "dp-echo-server", "-oyaml")
+			if err != nil {
+				return "", err
+			}
+			return output, nil
+		}, "10s", "1s").Should(ContainSubstring("health: {}"))
+
+		Eventually(func() (string, error) {
+			output, err := cluster.GetKumactlOptions().RunKumactlAndGetOutputV(Verbose, "get", "dataplane", "dp-demo-client", "-oyaml")
+			if err != nil {
+				return "", err
+			}
+			return output, nil
+		}, "10s", "1s").Should(ContainSubstring("ready: true"))
+	})
+
+})
```

Comment:
Nice! But here we are checking if we are updating the Dataplane definition.

For me the real E2E test would be to:
* spawn web
* spawn backend-1
* start stream of requests from web to backend
* spawn backend-2
* no requests are dropped, eventually requests are loadbalanced between backend-1 and backend-2.
* when service of backend-2 is down
* then eventually backend-2 is ejected and all requests are being sent to backend-1

Not a blocker for this PR to be merged, but I'd strongly consider extending the test to check such scenario.